### PR TITLE
Game statistics redo

### PR DESCRIPTION
**Pull request by priyapower**
Author: GROUP
Drivers: Chris & Priya

Created GameStatistics Class with

Method | Description | Return Value
-- | -- | --
highest_total_score | Highest sum of the winning and losing teams’ scores | Integer
lowest_total_score | Lowest sum of the winning and losing teams’ scores | Integer
percentage_home_wins | Percentage of games that a home team has won (rounded to the nearest 100th) | Float
percentage_visitor_wins | Percentage of games that a visitor has won (rounded to the nearest 100th) | Float
percentage_ties | Percentage of games that has resulted in a tie (rounded to the nearest 100th) | Float
count_of_games_by_season | A hash with season names (e.g. 20122013) as keys and counts of games as values | Hash
average_goals_per_game | Average number of goals scored in a game across all seasons including both home and away goals (rounded to the nearest 100th) | Float
average_goals_by_season | Average number of goals scored in a game organized in a hash with season names (e.g. 20122013) as keys and a float representing the average number of goals in a game for that season as values (rounded to the nearest 100th) | Hash

### CODE REVIEW

**Comment by lkriffell** [comment]
reviewed by: Logan Riffell

File: lib/game_statistics.rb [hunk 1]
```diff
@@ -1,104 +1,98 @@
...
+    total_score.min
   end
 
-  def percentage_visitor_wins
-    index = 0
-    visitor_wins = 0
-    data_size.times do
-      if @games_hash["away_goals"][index] > @games_hash["home_goals"][index]
-        visitor_wins += 1
+  def win_data
```
Comment (lkriffell) [inline lib/game_statistics.rb:33]: nice helper

File: lib/game_statistics.rb [hunk 2]
```diff
@@ -1,104 +1,98 @@
...
-    index = 0
-    tie_wins = 0
-    data_size.times do
-      if @games_hash["away_goals"][index] == @games_hash["home_goals"][index]
-        tie_wins += 1
-      end
-      index += 1
-    end
-    (tie_wins.to_f * 100 / data_size).round(2)
+  def percentage_of_home_wins
```
Comment (lkriffell) [inline lib/game_statistics.rb:45]: I see your percentage wins and ties

File: test/game_statistics_test.rb [hunk 3]
```diff
@@ -1,81 +1,47 @@
...
+  def test_it_can_calc_avg_goals_per_game
+    assert_equal 3.68, @game_statistics.average_goals_per_game
   end
 
-  def test_it_can_determine_percentage_ties
-    game_statistics = GameStatistics.new(@game.stat_hash)
-    assert_equal 5.26, game_statistics.percentage_ties
+  def test_it_can_calc_avg_goals_by_season
+    expected = {20122013=>3.68}
+    assert_equal expected, @game_statistics.average_goals_per_season
```
Comment (lkriffell) [inline test/game_statistics_test.rb:45]: I see all your tests

File: lib/game_statistics.rb (hunk 1 above)
Comment (priyapower) [reply lib/game_statistics.rb:33]: This was fun to write with you guys!

File: test/game_statistics_test.rb (hunk 3 above)
Comment (lkriffell) [reply test/game_statistics_test.rb:45]: I don't think all helper methods have a test. Add this to the refactor list

Comment (priyapower) [reply test/game_statistics_test.rb:45]: Oh wow! Great catch! Thanks for checking, will add to the todo for Refactoring: GameStatistics

**Pull request merged**